tests/test_isrTask.py

```diff
@@ -0,0 +1,550 @@
+            self.assertIsNotNone(results.opticsTransmission)
+            self.assertIsNotNone(results.filterTransmission)
+            self.assertIsNotNone(results.sensorTransmission)
+            self.assertIsNotNone(results.atmosphereTransmission)
+
+    def test_readIsrData_noTrans(self):
+        """Test that all necessary calibration frames are retrieved.
+        """
+        self.config.doAttachTransmissionCurve = False
+        self.task = IsrTask(config=self.config)
+        results = self.task.readIsrData(self.dataRef, self.inputExp)
+        self.validateIsrData(results)
+
+    def test_readIsrData_withTrans(self):
+        """Test that all necessary calibration frames are retrieved.
+        """
+        self.config.doAttachTransmissionCurve = True
+        self.task = IsrTask(config=self.config)
+        results = self.task.readIsrData(self.dataRef, self.inputExp)
+        self.validateIsrData(results)
+
+    def test_ensureExposure(self):
+        """Test that an exposure has a usable instance class.
+        """
+        self.assertIsInstance(self.task.ensureExposure(self.inputExp, self.camera, 0),
+                              afwImage.Exposure)
+
+    # def assertGreater(self, v1, v2):
+    #     import pdb
+    #     pdb.set_trace()
+    #     print(v1, ">", v2)
+    #     super().assertGreater(v1, v2)
+
+    # def assertLess(self, v1, v2):
+    #     import pdb
+    #     pdb.set_trace()
+    #     print(v1, "<", v2)
+    #     super().assertLess(v1, v2)
+
+    def test_convertItoF(self):
+        """Test conversion from integer to floating point pixels.
+        """
+        result = self.task.convertIntToFloat(self.inputExp)
+        self.assertEqual(result.getImage().getArray().dtype, np.dtype("float32"))
+        self.assertEqual(result, self.inputExp)
+
+    def test_updateVariance(self):
+        """Expect The variance image should have a larger median value after
+        this operation.
+        """
+        statBefore = computeImageMedianAndStd(self.inputExp.variance[self.amp.getBBox()])
+        self.task.updateVariance(self.inputExp, self.amp)
+        statAfter = computeImageMedianAndStd(self.inputExp.variance[self.amp.getBBox()])
+        self.assertGreater(statAfter[0], statBefore[0])
+        self.assertAlmostEqual(statBefore[0], 0.0, places=2)
+        self.assertAlmostEqual(statAfter[0], 8175.6885, places=2)
+
+    def test_darkCorrection(self):
+        """Expect the median image value should decrease after this operation.
+        """
+        darkIm = isrMock.DarkMock().run()
+
+        statBefore = computeImageMedianAndStd(self.inputExp.image[self.amp.getBBox()])
+        self.task.darkCorrection(self.inputExp, darkIm)
+        statAfter = computeImageMedianAndStd(self.inputExp.image[self.amp.getBBox()])
+        self.assertLess(statAfter[0], statBefore[0])
+        self.assertAlmostEqual(statBefore[0], 8075.6885, places=2)
+        self.assertAlmostEqual(statAfter[0], 8051.6206, places=2)
```

Comment:
I think we're better off using `assertFloatsAmostEqual` here, with `rtol` or `atol`: I find `assertAlmostEqual` with `places` to be an uncomfortable fit to our usage of numbers in LSST.